policy-ansible-for-nsxt/library/nsxt_vm_tags.py

```diff
@@ -0,0 +1,237 @@
+# A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: nsxt_vm_tags
+short_description: Update tags on NSXT VM
+description:
+    Update tags on NSXT VM
+version_added: "2.8"
+author: Gautam Verma
+options:
+    hostname:
+        description: Deployed NSX manager hostname.
+        required: true
+        type: str
+    username:
+        description: The username to authenticate with the NSX manager.
+        required: true
+        type: str
+    password:
+        description: The password to authenticate with the NSX manager.
+        required: true
+        type: str
+    validate_certs:
+        description: Enable server certificate verification.
+        type: bool
+        default: False
+    add_tags:
+        type: list
+        element: dict
+        description: List of tags to be applied to the virtual machine
+        suboptions:
+            scope:
+                description: Tag scope.
+                required: true
+                type: str
+            tag:
+                description: Tag value.
+                required: true
+                type: str
+    remove_tags_with_scope:
+        type: list
+        element: str
+        description:
+            - Specify the scope of the tags that should be removed
+            - If remove_other_tags is True, this becomes do not care
+    virtual_machine_id:
+        description: The identifier that is used in the enforcement point that
+                     uniquely identifies the virtual machine. In case of NSXT
+                     it would be the value of the external_id of the virtual
+                     machine.
+        type: str
+        required: true
+    remove_other_tags:
+        description:
+            - Remove the tags that are not specified in the add_tags
+            - Caution: If this is True, all tags that are not in add_tags will
+              be removed
+        default: false
+'''
+
+EXAMPLES = '''
+- name: Update Tags on VMs
+  nsxt_vm_tags:
+    hostname: "10.10.10.10"
+    username: "username"
+    password: "password"
+    validate_certs: False
+    virtual_machine_id: eaf6a927-552e-4285-8eaa-7b4f84bebca3
```

Comment:
Would it be possible to accept either virtual_machine_id OR virtual_machine_name? If it is name, we will then have to convert it to ID.

Reply:
added support